base/sparse/linalg.jl

```diff
@@ -585,42 +587,44 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
         S_old = copy(S)
         for j = 1:t
             for i = 1:n
-                S[i,j] = Y[i,j]==0?1:sign(Y[i,j])
+                S[i,j] = Y[i,j]==0?one(Y[i,j]):sign(Y[i,j])
             end
         end
 
-        # Check wether cols of S are parallel to cols of S or S_old
-        for j = 1:t
-            done = false
-            while ~done
-                repeated = false
-                if j > 1
-                    saux = S[1:n,j]' * S[1:n,1:j-1]
-                    for i = 1:j-1
-                        if abs(saux[i]) == n
-                            repeated = true
-                            break
+	if T <: Real
+            # Check wether cols of S are parallel to cols of S or S_old
+            for j = 1:t
+                done = false
+                while ~done
+                    repeated = false
+                    if j > 1
+                        saux = S[1:n,j]' * S[1:n,1:j-1]
+                        for i = 1:j-1
+                            if abs(saux[i]) == n
+                                repeated = true
+                                break
+                            end
                         end
                     end
-                end
-                if ~repeated
-                    saux2 = S[1:n,j]' * S_old[1:n,1:t]
-                    for i = 1:t
-                        if abs(saux2[i]) == n
-                            repeated = true
-                            break
+                    if ~repeated
+                        saux2 = S[1:n,j]' * S_old[1:n,1:t]
+                        for i = 1:t
+                            if abs(saux2[i]) == n
+                                repeated = true
+                                break
+                            end
                         end
                     end
-                end
-                if repeated
-                    for i = 1:n
-                        S[i,j] = rand()>=0.5?1:-1
+                    if repeated
+                        for i = 1:n
+                            S[i,j] = rand()>=0.5?1:-1
+                        end
+                    else
+                        done = true
                     end
-                else
-                    done = true
                 end
             end
-        end
+	end
```

Comment:
Not sure the indentation is correct here, it does not seem aligned with what follows. Could you check it, please?